oeps/oep-0013.rst

```diff
@@ -0,0 +1,576 @@
++---------------+-------------------------------------------+
+| Arbiter       |                                           |
++---------------+-------------------------------------------+
+| Status        | Draft                                     |
++---------------+-------------------------------------------+
+| Type          | Best Practice                             |
++---------------+-------------------------------------------+
+|  Created      | 2016-12-29                                |
++---------------+-------------------------------------------+
+| `Resolution`  |                                           |
++---------------+-------------------------------------------+
+| `Replaces`    |                                           |
++---------------+-------------------------------------------+
+| `Replaced-By` |                                           |
++---------------+-------------------------------------------+
+| `References`  |                                           |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+This document captures the conventions to be used for all Open edX web APIs.
+
+Motivation
+==========
+
+The Open edX platform contains many extension points for third-party developers that wish to integrate with the platform. We wish to approach a standard methodology for API design in order to give developers more consistency in their API client design, and to make our endpoint behaviors more predictable.
+
+Specification
+=============
+
+.. The specification describes the technical details of the Architecture, Best
+.. Practice, Process or Product Enhancement proposed by the OEP. If the proposal
+.. includes a new API, specify its syntax and semantics.
+
+High level considerations
+-------------------------
+
+**Consistent interface**
+
+All Open edX web APIs should use JSON as the data exchange format, and follow `RESTful API conventions <https://en.wikipedia.org/wiki/Representational_state_transfer>`_ where possible.
+
+
+**Consumer perspective**
+
+Design your API from the perspective of the consumer, NOT the perspective of your underlying implementation.  For example:
+
+    - If the underlying implementation requires accessing multiple models or multiple apps/projects, this does not need to be reflected in a public interface.  From the perspective of the consumer, it's simply one thing they are requesting.
+
+    - Keep CRUD operations together within its corresponding resource.  Why have the client go to one resource to read it and then another resource to write it?
+
+**Simple**
+
+Keep the top-leveI resources clear and simple - focusing on what the client is looking to consume.  You can hide complexity within the parameters.
+
+
+**Discoverable**
+
+Support `HATEOAS <https://en.wikipedia.org/wiki/HATEOAS>`_ where possible.  This allows us to change our URLs without needing to worry about updating the mobile apps if the app discovers its URLs from a base URL.
```

Comment:
Are we using HATEOAS in our newer APIs? I remember that @nasthagiri mentioned there were reasons why mobile doesn't use it. Is this still the recommendation? Does it change with the API gateway?